pkg/block/index.go

```diff
@@ -25,19 +25,22 @@ import (
 	"github.com/prometheus/tsdb/labels"
 )
 
-// IndexCacheFilename is the canonical name for index cache files.
-const IndexCacheFilename = "index.cache.json"
+const (
+	// IndexCacheVersion is a enumeration of index cache versions supported by Thanos.
+	IndexCacheVersion1 = iota + 1
+)
 
 type postingsRange struct {
 	Name, Value string
 	Start, End  int64
 }
 
 type indexCache struct {
-	Version     int
-	Symbols     map[uint32]string
-	LabelValues map[string][]string
-	Postings    []postingsRange
+	Version      int
+	CacheVersion int
```

Comment:
Not sure if this is OK or should it be opposite: `Version` for cache version and `IndexVersion` for index version.

Reply:
Also @GiedriusS if we want to add yours [JSON to bin/protobuf change](https://github.com/improbable-eng/thanos/pull/1013), should we start with that from the very cache version 1?

Reply:
I thought of it. Change could break some setups, if people preserve index cache permanently.

Reply:
Yup very good point - we want to be compatible. Thanks for being careful here :+1: